app/controllers/posts_controller.rb

```diff
@@ -1,27 +1,40 @@
 class PostsController < ApplicationController
+  before_action :set_post, only: %i[destroy update show]
   skip_before_action :authenticate_user_using_x_auth_token
   def index
     @posts = Post.includes(:categories, :user, :organization).order(created_at: :desc)
   end
 
   def new
-    @post = Post.new
+    post = Post.new
   end
 
   def create
-    post = Post.new(post_params)
-    if post.save
-      post.categories = Category.where(id: params[:category_ids])
-    end
+    post = Post.create(post_params)
```

Comment:
```suggestion
    post = Post.create!(post_params)
```

Read: https://courses.bigbinaryacademy.com/learn-rubyonrails/using-bang-method/